queries/vrl/highlights.scm

```diff
@@ -0,0 +1,110 @@
+(comment) @comment @spell
+
+(null) @constant.builtin
+
+(timestamp) @constant
+
+(closure_variables
+  (ident) @variable.parameter)
+
+(integer) @number
+
+(float) @number.float
+
+[
+  (string)
+  (raw_string)
+] @string
+
+[
+  (raw_string_escape_sequence)
+  (escape_sequence)
+  (regex_escape_sequence)
+] @string.escape
+
+(string_template
+  "{{" @punctuation.special
+  (_)
+  "}}" @punctuation.special)
+
+(regex) @string.regexp
+
+(boolean) @boolean
+
+(ident) @variable
+
+[
+  (null)
+  (noop)
+] @variable.builtin
+
+(query
+  (event) @variable.builtin)
+
+(function_call
+  (ident) @function.call)
+
+; VRL queries
+(query
+  (path
+    [
+      (field) @variable
+      (string) @string
+      (index) @number
+    ]))
+
+"return" @keyword.return
+
+"abort" @keyword.exception
+
+[
+  "if"
+  "else"
+] @keyword.conditional
+
+[
+  "="
+  "=="
+  "!="
+  "|="
+  ">"
+  ">="
+  "<"
+  "<="
+  "+"
+  "-"
+  "*"
+  "/"
+  "&&"
+  "||"
+  "??"
+  "|"
+  "!"
+] @operator
+
+[
+  "->"
+  ":"
+  ";"
+  ","
+] @punctuation.delimiter
+
+(path
+  "." @punctuation.delimiter)
```

Comment:
oh, and maybe check if you can include this in the `(query (path))` pattern above? (And same for the `(query ".")` pattern.)

Not necessary, but maybe at least bring them closer together.

Reply:
+1